integration/diff_05_document_validation_test.go

```diff
@@ -151,55 +118,84 @@ func TestDiffDocumentValidation(t *testing.T) {
 					return
 				}
 
-				if tc.werr != nil {
-					AssertEqualWriteError(t, *tc.werr, err)
-					return
-				}
-
-				AssertEqualCommandError(t, *tc.cerr, err)
+				AssertEqualWriteError(t, *tc.werr, err)
 			})
 		}
 	})
+}
 
-	t.Run("FindAndModify", func(t *testing.T) {
+func TestDiffDocumentValidationNaN(t *testing.T) {
```

Comment:
Connection is closed upon encountering NaN, so each test case has it's own setup.